v3/lint/base.go

```diff
@@ -125,6 +242,62 @@ func (l *Lint) execute(lint LintInterface, cert *x509.Certificate, config Config
 	} else if !l.CheckEffective(cert) {
 		return &LintResult{Status: NE}
 	}
-	res := lint.Execute(cert)
-	return res
+	return lint.Execute(cert)
+}
+
+// RevocationListLint represents a single x509 CRL linter.
+type RevocationListLint struct {
+	// Metadata associated with the linter.
+	LintMeta
+	// A constructor which returns the implementation of the linter.
+	Lint func() RevocationListLintInterface `json:"-"`
+}
+
+// CheckEffective returns true if r was generated on or after the EffectiveDate
+// AND before (but not on) the Ineffective date. That is, CheckEffective
+// returns true if...
+//
+//	r.ThisUpdate in [EffectiveDate, IneffectiveDate)
+//
+// If EffectiveDate is zero, then only IneffectiveDate is checked. Conversely,
+// if IneffectiveDate is zero then only EffectiveDate is checked. If both EffectiveDate
+// and IneffectiveDate are zero then CheckEffective always returns true.
+func (l *RevocationListLint) CheckEffective(r *ox509.RevocationList) bool {
+	onOrAfterEffective := l.EffectiveDate.IsZero() || util.OnOrAfter(r.ThisUpdate, l.EffectiveDate)
+	strictlyBeforeIneffective := l.IneffectiveDate.IsZero() || r.ThisUpdate.Before(l.IneffectiveDate)
+	return onOrAfterEffective && strictlyBeforeIneffective
+}
+
+// Execute runs the lint against a revocation list. For lints that are
+// sourced from the CA/B Forum Baseline Requirements, we first determine
```

Comment:
"For lints that are sourced from the CA/B Forum Baseline Requirements, we first determine if they are in within the purview of the BRs"

I think this was copied over, but isn't true of this code or may not be necessary for CRLs?

Reply:
Technically the execute operates pretty much the same way, but I do think that we don't need to mention CA/B here